[WIP] Release notes for 7.7.0 (#11721)
elastic/logstash: docs/static/releasenotes.asciidoc

e64c6f312 - Backport changes from 11544 (#11665) (3 weeks ago) <Karen Metts>Add more references to contributing issues #11544
229487c38 - [DOCS] Updates security API examples (#10752) (#11660) (3 weeks ago) <Lisa Cawley>
de95de3ae - remove TODO from CONTRIBUTING notes (4 weeks ago) <Karol Bucek>
17c11508d - Remove deprecation notices (4 weeks ago) <Karen Metts>
5587c9d7a - Make capitalization consistent (4 weeks ago) <Karen Metts>
239e9b0c3 - Fixes out-dated monitoring links (4 weeks ago) <lcawl>
9361638b2 - Add shared attribute for cloud trial (4 weeks ago) <Karen Metts>
80cd89673 - Fix missing " on glob pattern (8 weeks ago) <tbotalla>
5cbc9eeb4 - Add info about deprecation logger support mixin (8 weeks ago) <Karen Metts>
6a412f7af - Added precation notice when internal monitoring collector is used. Closes #11346 (8 weeks ago) <andsel>
23c7f48aa - Updated README to document how to run single Ruby specs (8 weeks ago) <andsel>
ca89cfee7 - Added section for monitoring.cluster_uuid (8 weeks ago) <andsel>
d2a094629 - Add cluster_uuid setting to default config file, displaying it in Node stats HTTP API (8 weeks ago) <Mike Place>

> 

Reviewer: ```suggestion

Monitoring - Additional option to send monitoring data directly to a dedicated monitoring cluster, rather than via the production cluster
  TODO: Flesh this out
* Conditionally add monitoring structure to top-level of output in API https://github.com/elastic/logstash/pull/11106[#11106]
* Backport of #11312 [put document_type for /_monitoring] https://github.com/elastic/logstash/pull/11640[#11640]
* Adjustments to send monitoring data directly to ES Backport of PR #11541 to 7.7 https://github.com/elastic/logstash/pull/11641[#11641]

Logging Improvements

* Added plugin.name to fish tag log lines https://github.com/elastic/logstash/pull/11078[#11078]
* Added plugin.id to docker images https://github.com/elastic/logstash/pull/11593[#11593]

> * Added pipeline.id to docker images https://github.com/elastic/logstash/pull/11567[#11567]

Reviewer: ```suggestion
This release continues with the logging improvements we have been making recently. This release adds the plugin name to log entries and includes the previous changes made to add plugin and pipeline identifiers to the docker image: https://github.com/elastic/logstash/pull/11078[#11078], https://github.com/elastic/logstash/pull/11593[#11593], https://github.com/elastic/logstash/pull/11567[#11567]
```

* Changed the default number of threads from `4` to `16` to match the default number from the Azure-Sdk EventProcessorHost https://github.com/logstash-plugins/logstash-input-azure_event_hubs/pull/54[#54]
* Fixed missing configuration of the `max_batch_size`setting https://github.com/logstash-plugins/logstash-input-azure_event_hubs/pull/52[#52]
* [DOC] Added clarification for threads parameter https://github.com/logstash-plugins/logstash-input-azure_event_hubs/pull/50[#50]

*Elasticsearch Input*

* Feat: added option to specify proxy for ES https://github.com/logstash-plugins/logstash-input-elasticsearch/pull/114[#114]

*S3 Input*

* Added support for including objects restored from Glacier or Glacier Deep https://github.com/logstash-plugins/logstash-input-s3/issues/199[#199]
* Added `gzip_pattern` option, enabling more flexible determination of whether a file is gzipped https://github.com/logstash-plugins/logstash-input-s3/issues/165[#165]
* Refactor: log exception: class + unify logging messages a bit https://github.com/logstash-plugins/logstash-input-s3/pull/201[#201]

*S3 Output*

> *  [DOC] Updated setting descriptions for clarity https://github.com/logstash-plugins/logstash-output-s3/pull/219[#219]and https://github.com/logstash-plugins/logstash-output-s3/pull/220[#220]

Reviewer: ```suggestion
*  [DOC] Updated setting descriptions for clarity https://github.com/logstash-plugins/logstash-output-s3/pull/219[#219] and https://github.com/logstash-plugins/logstash-output-s3/pull/220[#220]
```

* [DOC] Changed documentation to update the default number of threads https://github.com/logstash-plugins/logstash-input-azure_event_hubs/pull/55[#55]
* [DOC] Added clarification for threads parameter https://github.com/logstash-plugins/logstash-input-azure_event_hubs/pull/50[#50]

*Elasticsearch Input*

* Feature : Added option to specify proxy for Elasticsearch https://github.com/logstash-plugins/logstash-input-elasticsearch/pull/114[#114]

*S3 Input*

* Feature: Added support for including objects restored from Glacier or Glacier Deep https://github.com/logstash-plugins/logstash-input-s3/issues/199[#199]
* Feature: `gzip_pattern` option, enabling more flexible determination of whether a file is gzipped https://github.com/logstash-plugins/logstash-input-s3/issues/165[#165]
* Refactor: log exception: class + unify logging messages a bit https://github.com/logstash-plugins/logstash-input-s3/pull/201[#201]

*S3 Output*

* Feature: Added retry_count and retry_delay config. This allows opting out from infinite upload retries under error conditions. https://github.com/logstash-plugins/logstash-output-s3/pull/218[#218]

> * [DOC] Updated setting descriptions for clarity https://github.com/logstash-plugins/logstash-output-s3/pull/219[#219]and https://github.com/logstash-plugins/logstash-output-s3/pull/220[#220]

Reviewer: ```suggestion
* [DOC] Updated setting descriptions for clarity https://github.com/logstash-plugins/logstash-output-s3/pull/219[#219] and https://github.com/logstash-plugins/logstash-output-s3/pull/220[#220]
```

Reviewer: This fix got reverted in some of our back-and-forth.

  TODO: Flesh this out
** Conditionally add monitoring structure to top-level of output in API https://github.com/elastic/logstash/pull/11106[#11106]
** Backport of #11312 [put document_type for /_monitoring] https://github.com/elastic/logstash/pull/11640[#11640]
** Adjustments to send monitoring data directly to ES Backport of PR #11541 to 7.7 https://github.com/elastic/logstash/pull/11641[#11641]

* Feature: Logging Improvements
** This release continues with the logging improvements we have been making recently. This release adds the plugin name to log entries and includes the previous changes made to add plugin and pipeline identifiers to the docker image: https://github.com/elastic/logstash/pull/11078[#11078], https://github.com/elastic/logstash/pull/11593[#11593], https://github.com/elastic/logstash/pull/11567[#11567]

Feature: https://github.com/elastic/logstash/issues/10938[Previously], using the Java execution engine, there was no way to guarantee that ordering of event is preserved through a Logstash pipeline, even when setting `pipeline.workers` to  1 in logstash.yml, as is required by filters that require ordering to work correctly, such as the aggregation filter.

> This release adds a `pipeline.ordered` setting, which gives users the option to preserve event order when `pipeline.workers` is set to `1`. See *ADD_LINK_HERE* for more details. https://github.com/elastic/logstash/pull/11552[#11552], https://github.com/elastic/logstash/pull/11710[#11710]

Reviewer: ```suggestion
* Feature:  New Logstash monitoring option. +
We added a new option for using internal collectors to send monitoring data
directly to the monitoring cluster rather than the production cluster.
https://github.com/elastic/logstash/pull/11106[#11106],  
https://github.com/elastic/logstash/pull/11640[#11640],
https://github.com/elastic/logstash/pull/11641[#11641] 
+
Past implementations of internal collection sent monitoring data through a
production Elasticsearch cluster which forwarded the data to a dedicated
monitoring cluster. Now the internal collector can send monitoring data directly
to the monitoring cluster. 
+
Migrating from legacy internal collection to new
<<monitoring-internal-collection,Internal collection>> is straightforward. 

** Drop the `xpack.` prefix from your configuration settings. 
For example, `xpack.monitoring.enabled` is now `monitoring.enabled`. See
<<monitoring-settings, Monitoring Settings>> for the full list.
** Change the hosts setting to point to monitoring cluster instead of production
** If you don’t have an Elasticsearch output plugin configured in the pipelines,
add the setting monitoring.cluster_uuid to your logstash.yml.

+
See <<configuring-logstash, Monitoring Logstash>> for more information
about this and other monitoring options.

* Feature: Logging Improvements. +
This release continues with the logging improvements we have been making
recently. This release adds the plugin name to log entries and includes the
previous changes made to add plugin and pipeline identifiers to the docker
image:
https://github.com/elastic/logstash/pull/11078[#11078],
https://github.com/elastic/logstash/pull/11593[#11593],
https://github.com/elastic/logstash/pull/11567[#11567]

* Bugfix/Feature: The new `pipeline.ordered` setting gives users more control over
event ordering for single worker pipelines. 
+
Historically, {ls} events ordering between an input plugin and output plugin
were preserved when a single worker executed a pipeline. This behaviour was
never officially documented, but it was a feature that many users relied upon.
+
When the Java Execution engine was introduced, it broke the single worker
ordering preservation. This bug was fixed with the introduction of the
pipeline.ordered setting. 
https://github.com/elastic/logstash/pull/11552[#11552], 
https://github.com/elastic/logstash/pull/11710[#11710]
+
The `pipeline.ordered` setting is in <<logstash-settings-file,`logstash.yml`>>.
Valid options are `auto`, `true`, or `false`.
```

*Xml Filter*

* Feature: added parser_options to allow for more control of the parsing process. Includes the option of `strict` parsing to force the XML parser to fail early when parsing invalid XML.  https://github.com/logstash-plugins/logstash-filter-xml/pull/68[#68]

*Azure_event_hubs Input*

* Fixed: Now honors `max_batch_size` setting - previously this was ignored https://github.com/logstash-plugins/logstash-input-azure_event_hubs/pull/52[#52]
* Changed: The default number of threads is now `16` instead of `4`to match the default number from the Azure-Sdk EventProcessorHost https://github.com/logstash-plugins/logstash-input-azure_event_hubs/pull/54[#54]
* Refactor: scope and review global java_imports https://github.com/logstash-plugins/logstash-input-azure_event_hubs/pull/57[#57]
* [DOC] Changed documentation to update the default number of threads https://github.com/logstash-plugins/logstash-input-azure_event_hubs/pull/55[#55]
* [DOC] Added clarification for threads parameter https://github.com/logstash-plugins/logstash-input-azure_event_hubs/pull/50[#50]

*Elasticsearch Input*

> * Feature : Added option to specify proxy for Elasticsearch https://github.com/logstash-plugins/logstash-input-elasticsearch/pull/114[#114]

Reviewer: ```suggestion
* Feature: Added option to specify proxy for Elasticsearch https://github.com/logstash-plugins/logstash-input-elasticsearch/pull/114[#114]
```

* <<logstash-7-0-0-alpha2,Logstash 7.0.0-alpha2>>
* <<logstash-7-0-0-alpha1,Logstash 7.0.0-alpha1>>

[[logstash-7-7-0]]
=== Logstash 7.7.0 Release Notes

* Feature:  New Logstash monitoring option. +
We added a new option for using internal collectors to send monitoring data
directly to the monitoring cluster rather than the production cluster.
https://github.com/elastic/logstash/pull/11106[#11106],  
https://github.com/elastic/logstash/pull/11640[#11640],
https://github.com/elastic/logstash/pull/11641[#11641] 
+
Past implementations of internal collection sent monitoring data through a
production Elasticsearch cluster which forwarded the data to a dedicated
monitoring cluster. Now the internal collector can send monitoring data directly
to the monitoring cluster. 
+
Migrating from legacy internal collection to new
<<monitoring-internal-collection,Internal collection>> is straightforward. 

** Drop the `xpack.` prefix from your configuration settings. 
For example, `xpack.monitoring.enabled` is now `monitoring.enabled`. See
<<monitoring-settings, Monitoring Settings>> for the full list.

> ** Change the hosts setting to point to monitoring cluster instead of production

Reviewer: ```suggestion
** Change the hosts settings and configuration options to point to monitoring cluster instead of production
```
